packages/containers-shared/src/utils.ts

```diff
@@ -250,3 +243,88 @@ export async function checkExposedPorts(
 export function generateContainerBuildId() {
 	return randomUUID().slice(0, 8);
 }
+
+type DockerContext = {
+	Current: boolean;
+	Description: string;
+	DockerEndpoint: string;
+	Error: string;
+	Name: string;
+};
+
+/**
+ * Run `docker context ls` to get the Docker socket from the currently active Docker context
+ */
+export function getDockerSocketFromContext(dockerPath: string) {
+	try {
+		const output = runDockerCmdWithOutput(dockerPath, [
+			"context",
+			"ls",
+			"--format",
+			"json",
+		]);
+
+		// Parse each line as a separate JSON object
+		const lines = output.trim().split("\n");
+		const contexts: DockerContext[] = lines.map((line) => JSON.parse(line));
+
+		// Find the current context
+		const currentContext = contexts.find((context) => context.Current === true);
+
+		if (
+			currentContext &&
+			currentContext.DockerEndpoint &&
+			typeof currentContext.DockerEndpoint === "string"
+		) {
+			return currentContext.DockerEndpoint;
+		}
+	} catch {
+		// Fall back to null if docker context inspection fails
+	}
+	return null;
+}
+/**
+ * Resolve Docker host as follows:
+ * 1. Check WRANGLER_DOCKER_HOST environment variable
+ * 2. Check DOCKER_HOST environment variable
+ * 3. Try to get socket from active Docker context
+ * 4. Fall back to platform-specific defaults
+ */
+export function resolveDockerHost(dockerPath: string) {
```

Comment:
It would be nice to add an explicit return type / JSDoc here and for the next function